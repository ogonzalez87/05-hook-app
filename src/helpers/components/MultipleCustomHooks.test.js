import { shallow } from "enzyme";
import React from "react";
import { MultipleCustomHooks } from "../../components/03-examples/MultipleCustomHooks";
import { useFetch } from "../../hooks/useFetch";
jest.mock("../../hooks/useFetch");
describe("Multiple Hooks", () => {
  test("should match snapshoot", () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper).toMatchSnapshot();
  });

  test("should show info", () => {
    useFetch.mockReturnValue({
      data: [{ author: "fernando", quote: "hola mundo" }],
      loading: false,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    
    expect(wrapper.find(".alert").exists()).toBe(false);
    // expect(wrapper.find(".mb-0").text().trim()).toBe("HolaMundo");
    expect(wrapper.find("footer").text().trim()).toBe("fernando");
    expect(wrapper.find(".mb-3").text().trim()).toBe("hola mundo");

    // expect(wrapper).toMatchSnapshot();
  });
});
