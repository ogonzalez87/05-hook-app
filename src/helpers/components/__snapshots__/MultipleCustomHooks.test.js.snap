// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple Hooks should match snapshoot 1`] = `
<div>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente Quoute
  </button>
</div>
`;
